kartothek/io_components/write.py

```diff
@@ -72,7 +72,8 @@ def persist_common_metadata(partition_list, update_dataset, store, dataset_uuid)
             )
 
     result = {}
-    for table, schemas in tm_dct.items():
+    for table in sorted(tm_dct.keys()):
+        schemas = sorted(tm_dct[table], key=lambda s: s.origin)
```

Comment:
why sort it?

Reply:
reproducible error messages. Same input, same error, which I think is nice for users (and tests). I'll add a comment.